examples/nidaqmx_analog_input/teststand_nidaqmx.py

```diff
@@ -15,49 +19,42 @@ def create_nidaqmx_tasks(sequence_context: Any) -> None:
             (Dynamically typed.)
     """
     with GrpcChannelPoolHelper() as grpc_channel_pool:
-        session_management_client = nims.session_management.Client(
-            grpc_channel=grpc_channel_pool.session_management_channel
-        )
         teststand_support = TestStandSupport(sequence_context)
         pin_map_id = teststand_support.get_active_pin_map_id()
+        pin_map_context = PinMapContext(pin_map_id=pin_map_id, sites=None)
 
-        pin_map_context = nims.session_management.PinMapContext(pin_map_id=pin_map_id, sites=None)
-        grpc_device_channel = grpc_channel_pool.get_grpc_device_channel(
-            nidaqmx.GRPC_SERVICE_INTERFACE_NAME
+        discovery_client = DiscoveryClient(grpc_channel_pool=grpc_channel_pool)
+        session_management_client = SessionManagementClient(
+            discovery_client=discovery_client, grpc_channel_pool=grpc_channel_pool
         )
+
         with session_management_client.reserve_sessions(
-            context=pin_map_context,
-            instrument_type_id=nims.session_management.INSTRUMENT_TYPE_NI_DAQMX,
+            pin_map_context, instrument_type_id=INSTRUMENT_TYPE_NI_DAQMX
         ) as reservation:
-            for session_info in reservation.session_info:
-                task = create_task(
-                    session_info,
-                    grpc_device_channel,
-                    initialization_behavior=nidaqmx.SessionInitializationBehavior.INITIALIZE_SERVER_SESSION,
-                )
-                task.ai_channels.add_ai_voltage_chan(session_info.channel_list)
+            with reservation.create_nidaqmx_tasks(
+                initialization_behavior=SessionInitializationBehavior.INITIALIZE_SESSION_THEN_DETACH
+            ) as tasks:
+                for task in tasks:
+                    task.session.ai_channels.add_ai_voltage_chan(task.channel_list)
```

Comment:
FYI @csjall Unlike MI session creation, users need to be able to customize this aspect of DAQmx task creation.